components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -48,13 +73,33 @@ AIChatTabHelper::AIChatTabHelper(
           web_contents->GetBrowserContext()
               ->GetDefaultStoragePartition()
               ->GetURLLoaderFactoryForBrowserProcess(),
-          channel_name) {
+          channel_name),
+      pdf_load_observer_(
+          std::make_unique<PDFA11yInfoLoadObserver>(nullptr, this)) {
   favicon::ContentFaviconDriver::FromWebContents(web_contents)
       ->AddObserver(this);
 }
 
 AIChatTabHelper::~AIChatTabHelper() = default;
 
+void AIChatTabHelper::SetOnPDFA11yInfoLoadedCallbackForTesting(
+    base::OnceClosure cb) {
+  on_pdf_a11y_info_loaded_cb_ = std::move(cb);
+}
+
+void AIChatTabHelper::OnPDFA11yInfoLoaded() {
+  DVLOG(3) << "PDF Loaded";
+  is_pdf_a11y_info_loaded_ = true;
+  if (pending_get_page_content_callback_) {
+    FetchPageContent(web_contents(), "",
+                     std::move(pending_get_page_content_callback_));
```

Comment:
It was written as `DCHECK(!pending_get_page_content_callback_)` because `ConversationDriver` guarantees that there will only be once active request. However, I found out if we refresh the page while waiting for content fetching and click that "Summarize this page" again, we will enter the state that `GetPageContent called while another callback is pending`. 
Agree that it is better to use `base::OneShotEvent` if we want to wait for multiple callbacks.